### ISSUE TITLE
enh(ComparisonReport): Add pandas styling to the benchmark of metrics

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#1351]
### Is your feature request related to a problem? Please describe.

`ComparisonReport` has a `report_metrics` accessor that returns a pandas dataframe, this is used for benchmarking estimators

![Image](https://github.com/user-attachments/assets/c6a64809-a9df-4c99-b559-51baadb64cf0)

### Describe the solution you'd like

Add pandas styling to highlight for each metric, which estimator performs better

Something like below, but better:
- the gradient of color is per row
- less decimals
- take into account whether it's a score or an error

<img width="848" alt="Image" src="https://github.com/user-attachments/assets/d97363d9-c5d5-4c96-ba75-b38621811aa7" />

Suggestion of API:
```python
benchmark_metrics = comparator.metrics.report_metrics()  # as before
benchmark_metrics_styled = comparator.metrics.report_metrics(styling=True)  # as before
```

**Comment by MarieSacksick** [comment]
The API and the idea are good to me. However, I find the yellow background and red writing very agressive, and not a looking that you would use for something professional.  
I suggest using bold, to stick to what is used in research papers. I'm not sure we want to force specific colors because we want to display it correctly in any IDE, whatever the light or dark theme. @anasstarfaoui, any idea if we want to add colors?

**Comment by glemaitre** [comment]
Regarding colors, I would still stick to orange and blue :)

In my opinion, there are a couple of points to take into account:

### Precision of Digits

It's a great idea, but we should implement it correctly. With `CrossValidationReport`, we can compute the standard deviation of the metrics and deduce the number of relevant digits based on that.

If there is no such rationale, then I don't think we should modify the output.

### DataFrame vs. Style

The pandas Style object is not a DataFrame. We therefore cannot provide it to the user since they cannot use it afterwards. I think we should introduce the Style only in the `__repr__` method. The `__repr__` output is displayed to users but doesn't contain objects they need to work with.

**Comment by sylvaincom** [comment]
For the visuals, something like this?

Dark theme:
<img src="https://github.com/user-attachments/assets/706c055c-38bf-4251-865e-e50af4139995" width="600">

Light theme:
<img src="https://github.com/user-attachments/assets/ac7d1307-6099-4181-8d69-513329e22c61" width="600">

Even without cross-validation, I still think there are too many decimals (see [my comment](https://github.com/probabl-ai/skore/issues/1320#issuecomment-2674833333)), but not a strong opinion

**Comment by glemaitre** [comment]
I would advocate for the matplotlib color that are `tab:blue` and `tab:orange` and not probabl colors.

For the decimal, I need a rational to make a decision. If you have access to a source allowing to estimate the variance then I'm up for it. For instance, computing several scores via bootstrapping and making the decision if fine with me.

**Comment by glemaitre** [comment]
And it makes me think that having the option to compute interval based on bootstrap in the `report_metrics` of an `EstimatorReport` could be useful.

**Comment by sylvaincom** [comment]
Good point about matplotlib, then maybe no background color:

<img src="https://github.com/user-attachments/assets/758f07dd-e938-472c-9f64-45dd95f30108" width="600">

And just:

<img src="https://github.com/user-attachments/assets/de139aee-ee45-45ac-beec-1d0bfea0435e" width="600">

**Comment by sylvaincom** [comment]
<details>
<summary>Click to toggle contents of the source code to play with it yourself (choice of colors)</summary>

```python
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split

X, y = fetch_california_housing(return_X_y=True, as_frame=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline
from sklearn.linear_model import LinearRegression, Ridge
from sklearn.ensemble import RandomForestRegressor, HistGradientBoostingRegressor

estimators = [
    make_pipeline(StandardScaler(), LinearRegression()),
    make_pipeline(StandardScaler(), Ridge(random_state=0)),
    RandomForestRegressor(random_state=0),
    HistGradientBoostingRegressor(random_state=0),
]

from skore import EstimatorReport

estimator_reports = [
    EstimatorReport(est, X_train=X_train, X_test=X_test, y_train=y_train, y_test=y_test)
    for est in estimators
]
from skore import ComparisonReport

comparator = ComparisonReport(reports=estimator_reports)

benchmark_metrics = comparator.metrics.report_metrics()
benchmark_metrics
```

```python
my_df = comparator.metrics.report_metrics(scoring_names=["R2", "RMSE"])
my_df
```

```python
def highlight_best(s):
    score_metrics = ["R2"]
    is_max = s == s.max() if s.name in score_metrics else s == s.min()
    css_style_string = "color: #ff7f0e; font-weight: bold"
    return [css_style_string if v else "" for v in is_max]


styled_benchmark_metrics = (
    my_df
    .style.apply(highlight_best, axis=1)
)
styled_benchmark_metrics
```
</details>

**Comment by glemaitre** [comment]
If you want to go with gradient, then it makes sens that we only do that for the background (from white/black to orange/blue) where the color depend on the favorability of the metric. Basically, it would be a sort of report that looks like an heatmap.

The best model could be only a bold font.

**Comment by anasstarfaoui** [comment]
I think we should go for more smoother colors, even to highlight. These propositions seems a bit too harsh to the eye

**Comment by anasstarfaoui** [comment]
![Image](https://github.com/user-attachments/assets/02e2d074-1d91-4ccd-8b16-432053d6c15a)

![Image](https://github.com/user-attachments/assets/8de3d0c9-74a0-48cf-b157-db390fd0a89f)


I see two potential design for it, one associating blue with superior orange with inferior 

or simply all blue and consider that arrow pattern would be enough

**Comment by sylvaincom** [comment]
Many thanks! I like option 1

**Comment by MarieSacksick** [comment]
Let's go for the solution 1!  
Could you give us the hex values we should use for the max @anasstarfaoui please?  

Also: let's not forget to write the text in white for the dark blue cells. Otherwise, the other colors are quite pale, we can stick to black.

**Comment by MarieSacksick** [comment]
This issue should be done after https://github.com/probabl-ai/skore/issues/1359.

**Comment by MarieSacksick** [comment]
> @glemaitre @MarieSacksick do you have any suggestions for the API?

As @glemaitre said above:  
> The pandas Style object is not a DataFrame. We therefore cannot provide it to the user since they cannot use it afterwards. I think we should introduce the Style only in the __repr__ method. The __repr__ output is displayed to users but doesn't contain objects they need to work with.

To me, it means that there shouldn't really be a specific API. 
```python
display.frame() # displays a styled data frame
df = display.frame()
df # displays a usual dataframe
```
